shark_turbine/kernel/wave/wave.py

```diff
@@ -118,6 +137,25 @@ def is_reduction(node: fx.Node):
                 if tiling_constraint.dim == custom.axis:
                     tiling_constraint.induction_var = self.induction_vars[custom]
 
+    def initialize_wave_constraints(self, trace: CapturedTrace) -> None:
+        """
+        For each wave constraint, determines the appropriate wave id by looking
+        for workgroup constraints along the same dimension and using information
+        from the hardware constraints.
+
+        """
+
+        hardware_constraint = self.hardware_constraints[0]
+        thread_ids = [THREAD_0, THREAD_1, THREAD_2]
+        for wave_constraint in self.wave_constraints:
+            for workgroup_constraint in self.workgroup_constraints:
+                if wave_constraint.dim == workgroup_constraint.dim:
+                    wave_constraint.wave_id = thread_ids[
+                        workgroup_constraint.workgroup_dim
```

Comment:
Seems like we can infer the wave_id here, is there any reason to want to specify it from users? also wouldn't the user-specified wave_id actually be overriden if we do testExecute or invoke the `_trace_and_get_kernel_signature`?

Reply:
We don't ever want the user to specify the wave id. I use it only for testing. It should always be inferred.

Reply:
I se I see, that makes sense